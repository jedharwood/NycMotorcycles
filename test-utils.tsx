import React, { ReactElement } from 'react';

import { render, RenderOptions } from '@testing-library/react';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';
import { IntlProvider } from 'react-intl';
import { QueryClient, QueryClientProvider } from 'react-query';

import { langs, messages } from './types/languages';

type ProviderProps = {
    children: React.ReactNode;
    locale: string;
};

const AllTheProviders = ({ children, locale }: ProviderProps): JSX.Element => {
    const queryClient = new QueryClient();

    return (
        <MemoryRouterProvider>
            <QueryClientProvider client={queryClient}>
                <IntlProvider locale={locale} messages={messages[locale]}>
                    {children}
                </IntlProvider>
            </QueryClientProvider>
        </MemoryRouterProvider>
    );
};

type RenderWithProvidersOptions = RenderOptions & {
    locale?: string;
};

const renderWithProviders = (
    ui: ReactElement,
    { locale = langs.en, ...renderOptions }: RenderWithProvidersOptions = {}
) =>
    render(ui, {
        wrapper: ({ children }) => (
            <AllTheProviders locale={locale}>{children}</AllTheProviders>
        ),
        ...renderOptions,
    });

export * from '@testing-library/react';
export { renderWithProviders as render };
