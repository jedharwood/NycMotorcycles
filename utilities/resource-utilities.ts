export const yahooAuctionProfilePage: string =
  'https://auctions.yahoo.co.jp/seller/lazzamoto?'

export const instagramProfilePage: string =
  'https://www.instagram.com/newyorkcitymotorcycles/'

export const routes: { [key: string]: string } = {
  home: '/',
  activeAuctions: '/active-auctions',
  soldArchive: '/sold-archive',
  history: '/history',
  racing: '/racing',
  contact: '/contact',
  consign: '/consign',
  invest: '/invest',
  download: '/download',
}

const gallery: string = '/sold-archive/gallery/'
export const galleryRoutes: { [key: string]: string } = {
  hondaRc30: `${gallery}honda-rc30`,
  suzukiRg400: `${gallery}suzuki-rg400`,
  shovelheadChopper: `${gallery}shovelhead-chopper`,
  laverda750Sfc: `${gallery}laverda-750-sfc`,
  harleyXr750: `${gallery}harley-xr750`,
  kawasakiH2: `${gallery}kawasaki-h2`,
  suzukiStinger: `${gallery}suzuki-stinger`,
  bsaStarfire: `${gallery}bsa-starfire`,
  kawasakiH1: `${gallery}kawasaki-h1`,
}
