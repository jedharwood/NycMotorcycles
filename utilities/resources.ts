export const instagramLink: string =
    'https://www.instagram.com/newyorkcitymotorcycles/';
