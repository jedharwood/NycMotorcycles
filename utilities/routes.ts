const routes: { [key: string]: string } = {
  home: '/',
  activeAuctions: '/active-auctions',
  soldArchive: '/sold-archive',
  history: '/history',
  racing: '/racing',
  contact: '/contact',
  consign: '/consign',
  invest: '/invest',
  download: '/download',
  gallery: '/sold-archive/gallery',
};

export default routes;
