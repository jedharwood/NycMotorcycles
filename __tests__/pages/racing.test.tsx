import RacingPage from '@/pages/racing';
import { render } from '@/test-utils';
import { langs } from '@/utilities/resources';

describe('RacingPage', () => {
    it('should render page in English', () => {
        const { container } = render(<RacingPage />);

        expect(container).toMatchSnapshot();
    });

    it('should render page in Japanese', () => {
        const { container } = render(<RacingPage />, { locale: langs.ja });

        expect(container).toMatchSnapshot();
    });
});
