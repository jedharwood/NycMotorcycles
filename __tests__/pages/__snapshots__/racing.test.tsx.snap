// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RacingPage should render page in English 1`] = `
<div>
  <main>
    <div
      class="space-y-6"
    >
      <div
        class="relative"
      >
        <img
          alt="An image of Larry from New York City Motorcycles racing an orange, vintage Laverda motorcycle on a track."
        />
        <div
          class="absolute top-1/2 left-1/2 w-max max-w-[calc(100%-2rem)] -translate-x-1/2 -translate-y-1/2 rounded-md bg-stone-600 opacity-90 shadow-lg md:max-w-[calc(100%-3rem)]"
        >
          <h2
            class="inline-block px-2 py-2 text-center text-2xl text-stone-50 sm:px-8 sm:text-4xl md:text-5xl lg:text-6xl"
          >
            Racing
          </h2>
        </div>
      </div>
      <section
        class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
      >
        <div
          class="space-y-4 undefined"
        >
          <p>
            After attending the Barber Vintage Festival at the end of the 2013 race season as a spectator/collector/enthusiast, I boldly announced that 2013 would be the last year that I didn't race. And so it was.
          </p>
          <p>
            The first race in the 2014 AHRMA National Championship was at Roebling Road Raceway, in Savannah, GA.  After completing the AHRMA race school on the Friday, I was on the starting grid the following two days. By the end of the 2014 season I raced at 7 of the 10 weekend events, campaigning a heavily-modified ex-Tim Joyce T-100 unit Triumph in Sportsman 500 class and toward the end of the season, in addition, a Laverda 750SFC clone.
          </p>
        </div>
      </section>
      <section
        class="grid gap-4 sm:grid-cols-2 md:grid-cols-3"
      >
        <div
          class="rounded-md
            card-hover cursor-pointer 
            "
        >
          <img
            alt="An image of a cool looking vintage Harley Davidson motorcycle with an orange colour fairing and the number seven on the front."
          />
        </div>
        <div
          class="rounded-md
            card-hover cursor-pointer 
            "
        >
          <img
            alt="An image of Larry from New York City Motorcycles racing on a white coloured vintage motorcycle with the race number 667 on the front. He's leaning low into a corner."
          />
        </div>
        <div
          class="rounded-md
            card-hover cursor-pointer 
            "
        >
          <img
            alt="A black and white image a rider in full leathers on a vintage motorcycle. The leathers have the words 'Roper' and 'Vanson' on the back."
          />
        </div>
        <div
          class="rounded-md
            card-hover cursor-pointer 
            "
        >
          <img
            alt="An image of Larry from New York City Motorcycles racing on a vintage Laverda motorcycle. He's leaning low into a corner."
          />
        </div>
        <div
          class="rounded-md
            card-hover cursor-pointer 
            "
        >
          <img
            alt="An image of two riders on vintage motorcycles lining up for the start of a race wearing full leathers and orange helmets under a dramatic sky."
          />
        </div>
        <div
          class="rounded-md
            card-hover cursor-pointer 
            "
        >
          <img
            alt="An image of Larry from New York City Motorcycles racing a vintage Honda motorbike. He's riding at Willow Springs - the fastest road in the west."
          />
        </div>
      </section>
    </div>
  </main>
</div>
`;

exports[`RacingPage should render page in Japanese 1`] = `
<div>
  <main>
    <div
      class="space-y-6"
    >
      <div
        class="relative"
      >
        <img
          alt="placeholder"
        />
        <div
          class="absolute top-1/2 left-1/2 w-max max-w-[calc(100%-2rem)] -translate-x-1/2 -translate-y-1/2 rounded-md bg-stone-600 opacity-90 shadow-lg md:max-w-[calc(100%-3rem)]"
        >
          <h2
            class="inline-block px-2 py-2 text-center text-2xl text-stone-50 sm:px-8 sm:text-4xl md:text-5xl lg:text-6xl"
          >
            placeholder
          </h2>
        </div>
      </div>
      <section
        class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
      >
        <div
          class="space-y-4 undefined"
        >
          <p>
            placeholder
          </p>
          <p>
            placeholder
          </p>
        </div>
      </section>
      <section
        class="grid gap-4 sm:grid-cols-2 md:grid-cols-3"
      >
        <div
          class="rounded-md
            card-hover cursor-pointer 
            "
        >
          <img
            alt="placeholder"
          />
        </div>
        <div
          class="rounded-md
            card-hover cursor-pointer 
            "
        >
          <img
            alt="placeholder"
          />
        </div>
        <div
          class="rounded-md
            card-hover cursor-pointer 
            "
        >
          <img
            alt="placeholder"
          />
        </div>
        <div
          class="rounded-md
            card-hover cursor-pointer 
            "
        >
          <img
            alt="placeholder"
          />
        </div>
        <div
          class="rounded-md
            card-hover cursor-pointer 
            "
        >
          <img
            alt="placeholder"
          />
        </div>
        <div
          class="rounded-md
            card-hover cursor-pointer 
            "
        >
          <img
            alt="placeholder"
          />
        </div>
      </section>
    </div>
  </main>
</div>
`;
